aten/src/ATen/native/native_functions.yaml

```diff
@@ -530,7 +530,7 @@
 - func: dot_out(Tensor result, Tensor self, Tensor tensor) -> Tensor
   variants: function
 
-- func: einsum(std::string equation, TensorList tensors) -> Tensor
+- func: _einsum(std::string equation, TensorList tensors) -> Tensor
```

Comment:
can't we rename the Python wrapper in here, instead of renaming the function in ATen?

Reply:
I don't think the Python function should be called anything but einsum.
But we could keep the ATen name and use  torch._C.VariableFunctions.einsum from the Python wrapper and let the wrapper be torch.einsum in Python.

Reply:
I'm not sure if the functionality I was thinking is available at this level, but it was possible at some point to expose a different python name than the C name of the function, similar to [what is present here](https://github.com/pytorch/pytorch/blob/master/tools/autograd/deprecated.yaml#L7-L8).
So I was thinking about keeping the ATen name `einsum`, and having this Python wrapper be called `_einsum`, so that the rest of the PR doesn't change.